### ISSUE TITLE
[config] Compute short_path inside `config` object. Return as `config.short_path`

### ISSUE THREAD

**Issue by jgsogo** [conan-io/conan#6765]
That would be a feature/fix not only for Cygwin but for Windows too. If we forget about environment variables, this function creates the short_path in the same _drive_ as the input `path` parameter, and it can be different for every `path`.

I agree with you and we should probably compute the `short_home` value only once based on the cache folder/drive. Probably it is something to retrieve/store in the `config` object like `config.short_home` and forget about all the logic in this function. Probably better for another PR, don't you think so?

_Originally posted by @jgsogo in https://github.com/conan-io/conan/pull/6741/files_

**Comment by memsharded** [comment]
short_paths feature will be removed in Conan 2.0, so this is unlikely to be implemented for 1.X, lets close it.
